src/main/java/com/phoeniksoft/pickupbot/domain/advisor/BinaryAdvisor.java

```diff
@@ -0,0 +1,50 @@
+package com.phoeniksoft.pickupbot.domain.advisor;
+
+import com.phoeniksoft.pickupbot.domain.context.UserContext;
+import lombok.AllArgsConstructor;
+
+import static java.util.Objects.requireNonNull;
+
+@AllArgsConstructor
+public class BinaryAdvisor implements Advisor {
```

Comment:
@andrew-kuzmin It's the simplest possible implementation of the advisor. That regarding to the user answer (yes/no, helped/not, true/false - binary) goes left or right to the next advice. It's the start point. It will be modified and improved later.